src/main/java/com/wojciechbarwinski/demo/legendary_warehouse/services/OrderServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.wojciechbarwinski.demo.legendary_warehouse.services;
+
+
+import com.wojciechbarwinski.demo.legendary_warehouse.dtos.InsufficientStockDTO;
+import com.wojciechbarwinski.demo.legendary_warehouse.dtos.OrderDTO;
+import com.wojciechbarwinski.demo.legendary_warehouse.dtos.OrderLineDTO;
+import com.wojciechbarwinski.demo.legendary_warehouse.exceptions.InsufficientStockException;
+import com.wojciechbarwinski.demo.legendary_warehouse.exceptions.ProductNotFoundException;
+import com.wojciechbarwinski.demo.legendary_warehouse.stock.StockItem;
+import com.wojciechbarwinski.demo.legendary_warehouse.stock.StockRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private final StockRepository stockRepository;
+
+    public OrderServiceImpl(StockRepository stockRepository) {
+        this.stockRepository = stockRepository;
+    }
+
+    public void proceedOrder(OrderDTO order) {
+
+        checkIfProductsInStockAreAvailable(order);
+
+        removeQuantityOfItemsFromStock(order);
+    }
+
+    private void checkIfProductsInStockAreAvailable(OrderDTO order) {
```

Comment:
To sobie omówimy jak się zdzwonimy :D

Reply:
oki, jestem ciekaw o co chodzi. Bo pamiętam że sam mi o takich metodach mowiłeś